### PR TITLE
skip "already known transaction" errors

### PR DESCRIPTION
**Pull request by sunce86**
This PR skips node submission errors that are related to already known transactions.
Discussed here: https://cowservices.slack.com/archives/C0361CDD1FZ/p1648897295271429

### CODE REVIEW

File: crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs [hunk 1]
```diff
@@ -17,6 +17,15 @@ use futures::FutureExt;
 use gas_estimation::{EstimatedGasPrice, GasPrice1559};
 use shared::Web3;
 
+const ALREADY_KNOWN_TRANSACTION: [&str; 6] = [
```
Comment (nlordell) [inline crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs:20]: One thing you can do so you don't need to change the type if you add new messages is:
```suggestion
const ALREADY_KNOWN_TRANSACTION: &[&str] = &[
```

**Review by nlordell — APPROVED** [review]

File: crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs [hunk 2]
```diff
@@ -64,8 +73,17 @@ impl TransactionSubmitting for CustomNodesApi {
                     });
                 }
                 Err(err) => {
-                    tracing::warn!(?err, ?lable, "single custom node tx failed");
-                    super::track_submission_success(lable.as_str(), false);
+                    // error is not real error if transaction pool responded that received transaction is already in the pool
+                    let real_error = match err.clone() {
```
Comment (MartinquaXD) [inline crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs:77]: The error does not need to be cloned. This also works:
```rust
let real_error = match &err {
//...
```

File: crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs (hunk 1 above)
Comment (sunce86) [reply crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs]: Very nice. Thanks.

File: crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs (hunk 2 above)
Comment (sunce86) [reply crates/solver/src/settlement_submission/submitter/custom_nodes_api.rs]: True, thanks.

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**